env-var-docs/README.md

```diff
@@ -0,0 +1,91 @@
+# Environment Variable Documentation
+
+The main configuration file for the CiviForm server is in
+[application.conf](../server/conf/application.conf). Configuration values can
+be read at server start time from environment variables. This is done using the
+`${?SOME_ENV_VAR}` substitution syntax.
+
+To aid in deploying the CiviForm server, we ensure every environment variable
+referenced in the configuration file has appropriate documentation. This
+documentation lives in [env-var-docs.json](../server/conf/env-var-docs.json).
+See [./env_var_docs/schema/README.md](./env_var_docs/schema/README.md) for
+documentation on the format. As part of a CiviForm release, the documentation
+is rendered as markdown and added to our
+[documentation](https://docs.civiform.us/it-manual/sre-playbook/server-environment-variables).
+
+## Scripts
+
+[check_vars_documentation.py](./check_vars_documented.py) and
+[run_regex_tests.py](./run_regex_tests.py): runs on any PR editing
+application.conf or env-var-docs.json. They ensure the documentation is
+complete and that any provided regular expression tests pass.
+
+[generate_markdown.py](./generate_markdown.py): runs during our release
+automation. Generates markdown from env-var-docs.json and submits it to our
+[documentation
+repository](https://github.com/civiform/docs/tree/main/docs/it-manual/sre-playbook/server-environment-variables).
+
+## Developer setup
+
+Ensure you have a system installation of python version 3.9 or greater:
+
+```sh
+$ python3 --version
+```
+
+### Installing code dependencies
+
+Create a [virtual python
+environment](https://docs.python.org/3/library/venv.html). From the repository
+root, change your shell's working directory:
+
+```sh
+$ cd env-var-docs
+$ python3 -m venv venv
+```
+
+Activate the new virtual environment:
+
+```sh
+$ source venv/bin/activate
+```
+
+If you are ever unsure if you are using the system python installation or a
+virtual environment, run:
+
+```sh
+$ which python
+```
+
+The path should point to the created virtual environment in this directory.
+
+Install the dependencies:
+
+```sh
+$ pip install ./env_var_docs
+$ pip install -r requirements.txt
+```
+
+To exit the virtual environment, run:
+
+```sh
+$ deactivate
+```
```

Comment:
**suggestion (blocking):** instead of having a human read through and execute these steps please write a bin script that takes care of this

Reply:
I am hesitant adding a layer of indirection for the venv stuff, in both cases someone would have to run the activate and deactivate commands.  I think not wrapping these steps with bin scripts would make them easier to understand

Reply:
Could we compromise and add the bin script but leave the docs explaining how to do it without? I'm a big believer in speedy dev tooling

Reply:
Added a script that checks python 3.10+ is installed and creates a venv directory if one does not exist.  The activate and deactivate commands must be called in the interactive shell and do not work if called in a script that the shell calls.  The new bin script echos out the commands to run to make it easier tho

Reply:
perfect, thank you!